src/rootStore.ts

```diff
@@ -50,11 +50,14 @@ export const storesMap = new WeakMap<
 >()
 
 /**
- * Client-side application instance used for devtools
+ * Expose the client-side application instance used for devtools
  */
-export let clientApp: App | undefined /*#__PURE__*/
-export const setClientApp = (app: App) => (clientApp = app)
-export const getClientApp = () => clientApp
+let promise: Promise<App> | undefined
+let resolveApp: Function | undefined
+export const setClientApp = (app: App) =>
+  resolveApp && (resolveApp as Function)(app)
```

Comment:
with the change above the cast should be able to disappear
```suggestion
  resolveApp && resolveApp(app)
```